packages/notebook/src/actions.tsx

```diff
@@ -2076,6 +2076,24 @@ export namespace NotebookActions {
       }
     });
   }
+
+  /**
+   * Access last notebook history.
+   *
+   * @param notebook - The target notebook widget.
+   */
+  export function accessLastHistory(notebook: Notebook): any {
```

Comment:
```suggestion
  export function accessLastHistory(notebook: Notebook): void {
```

same below. But I really it feels like this should return `Promise<void>` - also see comments on implementation in `Notebook` class.

Reply:
I'm not sure I understand. What's the advantage of making these Promises?

Reply:
The advantage is that when a third-party code calls `NotebookActions.accessPreviousHistory()` (or respective command) they will be able to await it and only then perform another action. This is useful for downstream automation and performance testing. If the promise is dangling at this level we have no way to know when the action is done. Returning the promise rather than loosing it also makes it easier to unit test it.

Reply:
I probably should have been more clear this question I think depends on the above question that I asked in a different thread:

> Does it make sense to not have them be promises or async at all? As far as I can tell all the calls within the function are synchronous so maybe it makes sense to just return a string. with no await.

If I make the underlying back and forward calls regular synchronous functions, then there shouldn't be any dangling promise, right?

Reply:
Yes, but should they really by synchronous? On on hand requesting history is always async, on the other I see that the current implementation just requests it after each cell execution. This may be sub-optimal (because we generate overhead and wire traffic on each cell execution) and if the response is delayed may lead to subpar UX (being presented with outdated history). My thinking is that it should be async all along. This is, we would invalidate history on every execution but only request it when user asks for it. Then navigating within history can be from cache to improve response time.

Reply:
Oh I see. That makes a lot of sense. I made a push that should address that. Let me know what you think. I am still working on adding tests and docs though.